dashboard/app/helpers/locale_helper.rb

```diff
@@ -42,25 +44,15 @@ def options_for_locale_code_select
     options
   end
 
-  # Parses and ranks locale code strings from the Accept-Language header.
+  # Returns an Array of supported locale codes.
   def accepted_locales
-    header = request.env.fetch('HTTP_X_VARNISH_ACCEPT_LANGUAGE', '')
-    begin
-      locale_codes = header.split(',').map do |entry|
-        locale, weight = entry.split(';')
-        weight = (weight || 'q=1').split('=')[1].to_f
-        [locale, weight]
-      end
-      locale_codes.sort_by {|_, weight| -weight}.map {|locale, _| locale.strip}
-    rescue
-      Logger.warn "Error parsing Accept-Language header: #{header}"
-      []
-    end
+    I18n.available_locales.map(&:to_s)
   end
 
   # Strips regions off of accepted_locales.
   def accepted_languages
-    accepted_locales.map {|locale| locale.split('-')[0]}
+    accepted_languages = accepted_locales.map {|locale| locale.split('-')[0]}
+    accepted_languages.uniq
```

Comment:
I would suggest modifying `#language` a little bit and using it in `#accepted_languages` to  point out on the dependency between the two methods.
```ruby
def language(locale_code = locale)
  locale_code.to_s.split('-').first
end
```

```suggestion
    accepted_languages = accepted_locales.map(&method(:language)).uniq
```

Reply:
Memoize it one to not collect these locales on each call
```suggestion
    @accepted_languages ||= accepted_locales.map(&method(:language)).uniq
```

Reply:
Thanks! I looked at this [memoization article](https://www.honeybadger.io/blog/ruby-rails-memoization/), which was useful for understanding how it works in the background.